lib/chef/resource/launchd.rb

```diff
@@ -30,16 +30,39 @@ class Launchd < Chef::Resource
       default_action :create
       allowed_actions :create, :create_if_missing, :delete, :enable, :disable, :restart
 
-      property :label, String, identity: true, name_property: true
-      property :backup, [Integer, FalseClass], desired_state: false
-      property :cookbook, String, desired_state: false
-      property :group, [String, Integer]
-      property :plist_hash, Hash
-      property :mode, [String, Integer]
-      property :owner, [String, Integer]
-      property :path, String
-      property :source, String
-      property :session_type, String
+      property :label, String,
+               identity: true, name_property: true,
+               description: "The unique identifier for the job."
+
+      property :backup, [Integer, FalseClass],
+               desired_state: false,
+               description: "The number of backups to be kept in /var/chef/backup. Set to false to prevent backups from being kept."
+
+      property :cookbook, String,
+               desired_state: false,
+               description: "The name of the cookbook in which the source files are located."
+
+      property :group, [String, Integer],
+               description: "When launchd is run as the root user, the group to run the job as. If the username property is specified and this property is not, this value is set to the default group for the user."
+
+      property :plist_hash, Hash,
+               introduced: "12.19",
+               description: "A Hash of key value pairs used to create the launchd property list."
+
+      property :mode, [String, Integer],
+               description: "A quoted 3-5 character string that defines the octal mode. For example: '755', '0755', or 00755."
```

Comment:
It appears this takes a integer too, and `quoted` is odd (makes it sound like you need to do `"'755'"` or something.